conf/resources.config

```diff
@@ -44,127 +34,42 @@ process {
-        cpus   = { check_max( 2     * task.attempt, 'cpus'    ) }
-        memory = { check_max( 8.GB  * task.attempt, 'memory'  ) }
-        time   = { check_max( 4.h   * task.attempt, 'time'    ) }
-    }
-
-    withName: 'NFCORE_CUTANDRUN:CUTANDRUN:SEACR_CALLPEAK' {
-        cpus   = { check_max( 2     * task.attempt, 'cpus'    ) }
-        memory = { check_max( 8.GB  * task.attempt, 'memory'  ) }
-        time   = { check_max( 4.h   * task.attempt, 'time'    ) }
-    }
-
-    withName: 'NFCORE_CUTANDRUN:CUTANDRUN:FASTQC_TRIMGALORE:FASTQC' {
-        cpus   = { check_max( 2     * task.attempt, 'cpus'    ) }
-        memory = { check_max( 8.GB  * task.attempt, 'memory'  ) }
-        time   = { check_max( 4.h   * task.attempt, 'time'    ) }
-    }
-
-    // memory
-    withName: 'NFCORE_CUTANDRUN:CUTANDRUN:DEEPTOOLS_PLOTHEATMAP_PEAKS' {
-        cpus   = { check_max( 1     * task.attempt, 'cpus'    ) }
-        memory = { check_max( 32.GB  * task.attempt, 'memory'  ) }
-        time   = { check_max( 4.h   * task.attempt, 'time'    ) }
+if(params.run_reporting) {
+    process {
+        withName: 'NFCORE_CUTANDRUN:CUTANDRUN:FRAG_LEN_HIST' {
+            cpus   = { check_max( 1     * task.attempt, 'cpus'    ) }
+            memory = { check_max( 32.GB  * task.attempt, 'memory'  ) }
+            time   = { check_max( 4.h    * task.attempt, 'time'    ) }
+        }
     }
+}
 
-    withName: 'NFCORE_CUTANDRUN:CUTANDRUN:.*:SAMTOOLS_VIEW' {
-        cpus   = { check_max( 1     * task.attempt, 'cpus'    ) }
-        memory = { check_max( 32.GB  * task.attempt, 'memory'  ) }
-        time   = { check_max( 4.h   * task.attempt, 'time'    ) }
+if(params.run_reporting && params.run_deeptools_heatmaps) {
+    process {
+        withName: 'NFCORE_CUTANDRUN:CUTANDRUN:DEEPTOOLS_COMPUTEMATRIX_GENE' {
+            cpus   = { check_max( 16     * task.attempt, 'cpus'    ) }
+            memory = { check_max( 64.GB  * task.attempt, 'memory'  ) }
+            time   = { check_max( 4.h    * task.attempt, 'time'    ) }
+        }
+
+        withName: 'NFCORE_CUTANDRUN:CUTANDRUN:DEEPTOOLS_COMPUTEMATRIX_PEAKS' {
+            cpus   = { check_max( 16     * task.attempt, 'cpus'    ) }
+            memory = { check_max( 64.GB  * task.attempt, 'memory'  ) }
+            time   = { check_max( 4.h    * task.attempt, 'time'    ) }
+        }
+
+        withName: 'NFCORE_CUTANDRUN:CUTANDRUN:DEEPTOOLS_PLOTHEATMAP_GENE' {
+            cpus   = { check_max( 1     * task.attempt, 'cpus'    ) }
+            memory = { check_max( 64.GB  * task.attempt, 'memory'  ) }
+            time   = { check_max( 4.h   * task.attempt, 'time'    ) }
+        }
+
+        withName: 'NFCORE_CUTANDRUN:CUTANDRUN:DEEPTOOLS_PLOTHEATMAP_PEAKS' {
+            cpus   = { check_max( 1     * task.attempt, 'cpus'    ) }
+            memory = { check_max( 64.GB  * task.attempt, 'memory'  ) }
+            time   = { check_max( 4.h   * task.attempt, 'time'    ) }
+        }
```

Comment:
I think it could be good to uncouple that part from `modules.config` and put it in `base.config`

Reply:
Because base config is part of the nf-core template and the other file is my own custom one. Its just style really, if it was a big deal I can include in the base config